### PR TITLE
Remove unused APIs on BugsnagEvent interface

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal

Removes/hides unused APIs from the `BugsnagEvent` interface. These are either no longer required or were originally part of the internal implementation and mistakenly included in the public header.

## Changeset

The following APIs have been removed/hidden using a [class extension](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/ProgrammingWithObjectiveC/CustomizingExistingClasses/CustomizingExistingClasses.html#//apple_ref/doc/uid/TP40011210-CH6-SW6):

```
BSGParseSeverity
BSGFormatSeverity
serializableValueWithTopLevelData
shouldBeSent
toJson
attachCustomStacktrace:withType:
enhancedErrorMessageForThread
notifyReleaseStages
handledState
overrides
depth
error
isIncomplete
```

## Tests

Updated existing test coverage to avoid using deprecated APIs. Mazerunner scenarios which were testing individual stackframes have had these assertions removed as historically this could be quite flaky.

### CODE REVIEW

File: features/fixtures/ios-swift-cocoapods/iOSTestApp/scenarios/HandledInternalNotifyScenario.swift [hunk 1]
```diff
@@ -19,7 +19,6 @@ import Bugsnag
                 ["method":"bar()", "file":"parser.js"],
                 ["method":"yes()"]
             ]
-            report.attachCustomStacktrace(frames, withType: "unreal")
```
Comment (snmaynard) [inline features/fixtures/ios-swift-cocoapods/iOSTestApp/scenarios/HandledInternalNotifyScenario.swift]: there is a maze runner expectation on this type that is failing

File: features/fixtures/ios-swift-cocoapods/iOSTestApp/scenarios/UnhandledInternalNotifyScenario.swift [hunk 2]
```diff
@@ -19,7 +19,6 @@ import Bugsnag
                 ["method":"baz()", "file":"[native code]"],
                 ["method":"is_done()"]
             ]
-            report.attachCustomStacktrace(frames, withType: "fake")
```
Comment (snmaynard) [inline features/fixtures/ios-swift-cocoapods/iOSTestApp/scenarios/UnhandledInternalNotifyScenario.swift]: there is a maze runner expectation on this that is failing

File: features/fixtures/ios-swift-cocoapods/iOSTestApp/scenarios/HandledInternalNotifyScenario.swift (hunk 1 above)
Comment (fractalwrench) [reply features/fixtures/ios-swift-cocoapods/iOSTestApp/scenarios/HandledInternalNotifyScenario.swift]: Yep - left this in the draft state overnight as the mazerunner scenario was using one of the removed APIs. Planning on updating the scenario/assertions today.

**Comment by tomlongridge** [comment]
It looks like we've had to remove some of the test rigor in this change now that we've made some field private. Can we not use some class extensions to give access in the tests?

**Comment by fractalwrench** [comment]
After IRL discussion `attachCustomStacktrace` has been reinstated as it was before, as this allows us to retain the same test coverage. It will be removed as part of future work once the Error interface has been implemented and the existing scenarios can be updated to use this API.

**Pull request merged**